From 1d55fa64eff459e544d39bd24b630bc9da046269 Mon Sep 17 00:00:00 2001
From: Josh Wilsdon <josh@wilsdon.ca>
Date: Wed, 21 Nov 2018 12:23:54 -0800
Subject: [PATCH] TRITON-976 agents-installer should have option to just unpack
 shar without executing install.sh Reviewed by: Trent Mick <trentm@gmail.com>
 Approved by: Trent Mick <trentm@gmail.com>

---
 mk-agents-shar | 15 ++++++++++++---
 package.json   |  2 +-
 2 files changed, 13 insertions(+), 4 deletions(-)

diff --git a/mk-agents-shar b/mk-agents-shar
index 7158859..25fe0e8 100755
--- a/mk-agents-shar
+++ b/mk-agents-shar
@@ -6,7 +6,7 @@
 #
 
 #
-# Copyright (c) 2014, Joyent, Inc.
+# Copyright (c) 2018, Joyent, Inc.
 #
 
 #
@@ -219,12 +219,17 @@ function mk_agents_shar() {
 
     (cat <<__EOF__
 #!/bin/sh
-if [ \`pwd\` != '$tmp' ]; then
-  cd $tmp
+
+if [[ -n \$AGENTSSHAR_UNPACK_DIR ]]; then
+    mkdir -p \$AGENTSSHAR_UNPACK_DIR
+    cd \$AGENTSSHAR_UNPACK_DIR
+elif [[ \`pwd\` != '$tmp' ]]; then
+    cd $tmp
 fi
 
 # Clean up possible previous runs
 rm -fr agents
+
 __EOF__
 )> $sh_filename
 
@@ -233,6 +238,10 @@ __EOF__
 set -o errexit
 set -o pipefail
 
+if [[ \$AGENTSSHAR_UNPACK_ONLY == "true" ]]; then
+    exit 0
+fi
+
 if [[ -f agents/install.sh ]]; then
     (cd agents && /bin/bash ./install.sh)
 fi
diff --git a/package.json b/package.json
index 7000262..9c289b3 100644
--- a/package.json
+++ b/package.json
@@ -1,7 +1,7 @@
 {
   "name": "agents-installer",
   "description": "Build agentsshar package for SDC",
-  "version": "2.1.0",
+  "version": "2.2.0",
   "author": "Joyent (joyent.com)",
   "private": true,
   "dependencies": {
-- 
2.21.0

